src/normalize.jl

```diff
@@ -0,0 +1,56 @@
+using LinearAlgebra
+
+function LinearAlgebra.normalize(tn::AbstractITensorNetwork; alg="exact", kwargs...)
+  return normalize(Algorithm(alg), tn; kwargs...)
+end
+
+function LinearAlgebra.normalize(alg::Algorithm"exact", tn::AbstractITensorNetwork)
+  norm_tn = norm_sqr_network(tn)
+  log_norm = logscalar(alg, norm_tn)
+  tn = copy(tn)
+  L = length(vertices(tn))
+  c = exp(log_norm / L)
+  for v in vertices(tn)
+    tn[v] = tn[v] / sqrt(c)
+  end
+  return tn
+end
+
+function LinearAlgebra.normalize(
+  alg::Algorithm"bp",
+  tn::AbstractITensorNetwork;
+  (cache!)=nothing,
+  update_cache=isnothing(cache!),
+  cache_update_kwargs=default_cache_update_kwargs(cache!),
+)
+  if isnothing(cache!)
+    cache! = Ref(BeliefPropagationCache(QuadraticFormNetwork(tn)))
+  end
+
+  if update_cache
+    cache![] = update(cache![]; cache_update_kwargs...)
+  end
+
+  tn = copy(tn)
+  cache![] = normalize_messages(cache![])
+  norm_tn = tensornetwork(cache![])
+
+  vertices_states = Dictionary()
+  for v in vertices(tn)
+    v_ket, v_bra = ket_vertex(norm_tn, v), bra_vertex(norm_tn, v)
+    pv = only(partitionvertices(cache![], [v_ket]))
+    vn = region_scalar(cache![], pv)
+    state = (1.0 / sqrt(vn)) * tn[v]
```

Comment:
```suggestion
    state = tn[v] / sqrt(vn)
```
`1.0` is implicitly of type `Float64` in Julia, so what you wrote would promote from single precision to double precision (say if you want to use `Float32`, for example on GPU).